src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -238,5 +240,25 @@ public void Dispose()
         {
             this.SavePeers();
         }
+
+        /// <summary>
+        /// When the PeerAddressManager saves peers to disk, check if any of
+        /// banned Peers have expired.  If so, then reset the Peers.
+        /// </summary>
+        private void ResetBannedPeers()
```

Comment:
We probably dont need to reset banned peers we just explode banned peers in the selection algorithm.

Reply:
The reset was to make sure the data persisted to the Peers.json file is correct.  So if the BanUntil has passed we need a way null out BanDate and BanUntil in the json file.

Reply:
hmm ok I see, it just to reduce the size of the file where do we call this from?

Reply:
When we load the peers.json file - PeerAddressManager.LoadPeers().

Reply:
Ok so if we call this both onLoad and onSave this is fine I guess